includes/blocks/class-sensei-course-outline-block.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * File containing the Sensei_Course_Outline_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_Outline_Block
+ */
+class Sensei_Course_Outline_Block {
+	/**
+	 * Sensei_Course_Outline_Block constructor.
+	 */
+	public function __construct() {
+		add_action( 'enqueue_block_assets', [ $this, 'enqueue_assets' ] );
+		add_action( 'init', [ $this, 'register_course_template' ], 101 );
+	}
+
+	/**
+	 * Enqueue frontend and editor assets.
+	 *
+	 * @access private
+	 */
+	public function enqueue_assets() {
+		if ( 'course' !== get_post_type() ) {
```

Comment:
I'm not totally comfortable with this solution. But it seems there is not a better way still. There's a couple of issues related to that, but any with a solution.

For our case, I also explored the idea to not list the block (`settings.support.inserter = false`) and have the post template locked (can't add/remove blocks), but it would happen for all blocks in the post and if the user wanted to remove it for some reason, they couldn't too.